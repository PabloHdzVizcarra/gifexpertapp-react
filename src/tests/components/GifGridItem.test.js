import React from 'react';
import GifGridItem from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('tests del componente <GifGridItem />', () => {

  const title = 'Este seria el titulo';
  const url = 'http://algunaImagen.jpg ';
  const wrapper = shallow(<GifGridItem title={title} url={url}/>);
  
  test('el componente GifGridItem debe renderizarse correctamente', () => {
    
    expect(wrapper).toMatchSnapshot();
  })

  test('debe de tener un parrafo con el title ', () => {
    
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  })

  test('la imagen debe tener el url y title de props', () => {
    
    const img = wrapper.find('img');
    // console.log(img.html()); te mostrara todo el html del elemento seleccionado
    // console.log(img.props().src);

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);

  })
  
  test('debe de tener la clase animate__fadeIn', () => {
    
    const div = wrapper.find('div');
    // console.log(div.html());
    // console.log(div.props());
    // console.log(div.prop('className').includes('animate__fadeIn'));

    expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
  })
  // 
  
})

