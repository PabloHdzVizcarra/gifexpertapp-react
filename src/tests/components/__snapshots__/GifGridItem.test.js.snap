// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests del componente <GifGridItem /> el componente GifGridItem debe renderizarse correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Este seria el titulo"
    src="http://algunaImagen.jpg "
  />
  <p>
    Este seria el titulo
  </p>
</div>
`;
